_posts/2020-05-11-BART-Summary.md

```diff
@@ -0,0 +1,73 @@
+
+## **How does it work?** 
+
+### **The Framework**
+![image](https://lh3.googleusercontent.com/_ZYOOgt3efQF8LlFM_rmlJdiQyj3bkFeKfeihhbOK3w-UvPUPvFX9K_YFMh7SIURsyFclNwkL8oVByH3XlQKXPnhZYO8IFY54nhFBlE9wuk0vJBKxI1Ci_7xnbePqT8thQC-vB1ZUvs)
+The idea behind the framework the authors propose is simple, they suggest that decoupling Language Models and the functions with which we corrupt the text is useful to compare different pre-training techniques and see how they fare on similar models and diverse benchmarks. Viewed this way pre-training is a sequence of repeated steps 
+* We apply a noising function to the text 
+* The language model attempts to reconstruct the text
+* We calculate the Loss Function, typically CrossEntropy over the original text, we back-propagate the gradients and update the model's weights.
+        
+### **Comparing different Text-Noising techniques and LM Objectives**
+![](https://firebasestorage.googleapis.com/v0/b/firescript-577a2.appspot.com/o/imgs%2Fapp%2FAntonioLprd%2FdqcEKBSd0Y.png?alt=media&token=2cee8cc9-cb39-472f-9377-82c995a7ee85)
+In the first experiment detailed in the paper Lewis et al., using the framework they introduced at the beginning of the article, compare different pre-training techniques and LM Objectives on a smaller than usual model, BART-base. The model uses a 6 layered, transformer-based, Seq2Seq architecture for autoencoding as introduced in the **"Attention is all you need"** paper from Vaswani et al. The pre-training techniques compared in the experiments can be divided between those that work at the token level and the ones that work at the sentence level.
+
+**Token Masking** random tokens are sampled and replaced with [MASK]<br />
+**Token Deletion** similar to masking but the token sampled are deleted and the model has to add new token in their place.<br />
+**Token Infilling** a number of text spans, i.e. contiguous group tokens, are sampled, and then they are replaced by the [MASK] token.<br />
+**Sentence Permutation** random shuffling of the document’s sentences.<br />
+**Document Rotation** a token is chosen randomly to be the start of the document, the section before the starting token is appended at the end.<br />
+
+Intuitively the techniques that work at the sentence level should help the LM learn the different roles of sentences in a paragraph or longer text and so help in NLG tasks.
+
+Besides the pre-training techniques, the authors also compare different LM objectives focusing on the ones used by BERT and GPT and also the ones that tried to get the best of both worlds.
+
+**Autoregressive, left to right, LM** (GPT-2)<br />
+**Masked LM**(BERT) replace 15% of the token with [MASK] and predict the corresponding words.<br />
+**Permuted LM** (XLNet) left to right, autoregressive LM training but with the order of the words to predict chosen at random. (worth looking into more, very interesting approach)<br />
```

Comment:
Is the parenthasis thought as a comment to the reader or as a reminder for you? If its the latter I think it can be removed, otherwise it might be helpful to provide some links to related readings etc.